### PR TITLE
CSVDiff to SchemaDiff

### PR DESCRIPTION
**Pull request by milljm**
Use DeepDiff for diffing routines in the TestHarness.

Unable to push to original repository, creating a new PR.

Refs# 23522

### CODE REVIEW

File: test/tests/test_harness/csvdiff_tests [hunk 1]
```diff
@@ -95,4 +95,11 @@
     override_rel_err = 5.5e-06
     requirement = 'spec file shall control a specific fields absolute zero tolerance'
   []
+  [test_csvdiff_comparison_custom_column]
+    type = CSVDiff
+    input = csvdiff_comparison_custom_column.i
+    csvdiff = csvdiff_comparison_custom_column_out.csv
+    custom_columns = time
+    requirement = 'CSVDiff shall compare only a specific column'
```
Comment (lindsayad) [inline test/tests/test_harness/csvdiff_tests:103]:
```suggestion
    requirement = 'The system shall be able to diff CSV files for only a specific column.'
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
Maybe a person who does python more than I do should review this 😆

**Review by permcody — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

**Comment by dschwen** [comment]
We don't have any tests that actually make sure that a tester that is meant to fail will fail, do we?

**Comment by loganharbour** [comment]
> We don't have any tests that actually make sure that a tester that is meant to fail will fail, do we?

Nope. But we will now.

**Pull request merged**